src/adapter/src/coord.rs

```diff
@@ -2970,25 +2970,34 @@ impl Coordinator {
                 .system_config()
                 .coord_slow_message_warn_threshold();
 
+            // How many messages we'd like to batch up before processing them. Must be > 0.
+            const MESSAGE_BATCH: usize = 64;
+            let mut messages = Vec::with_capacity(MESSAGE_BATCH);
+            let mut cmd_messages = Vec::with_capacity(MESSAGE_BATCH);
+
+            let message_batch = self.metrics
+                .message_batch
+                .with_label_values(&[]);
+
             loop {
                 // Before adding a branch to this select loop, please ensure that the branch is
                 // cancellation safe and add a comment explaining why. You can refer here for more
                 // info: https://docs.rs/tokio/latest/tokio/macro.select.html#cancellation-safety
-                let msg = select! {
+                select! {
                     // Order matters here. Some correctness properties rely on us processing
                     // internal commands before processing external commands.
```

Comment:
Can you update this comment to mention the batching? That if a user command generates any internal commands, it will not get processed until all existing queued commands are processed?

Reply:
Done.